server/routes/wishCards.js

```diff
@@ -411,48 +421,198 @@ router.post(
+
+    // in case user doesn't confirm donation, check after countdown runs out
+    blockedWishcardsTimer[wishCardId] = setTimeout(async () => {
+      queue.add({wishCardId,
+        userId,
+        url:wishCard.wishItemURL,
+        price: wishCard.wishItemPrice});
+    }, process.env.WISHCARD_LOCK_IN_MINUTES*1000*60);
+
+    res.status(200).send({
+      lockedUntil: lockedWishCard.isLockedUntil,
+      success: true,
+      error: null,
+    });
+  } catch (error) {
+    handleError(res, 400, error);
+  }
+});
+
+
+router.post('/unlock/:id', async (req, res) => {
+  try {
+
+    const {wishCardId, lockedWishcard, userId, error} = await getLockedWishCards(req);
+
+    if (error) handleError(res, 400, error)
+
+    if(lockedWishcard && lockedWishcard.isLockedBy === userId) {
+
+      await WishCardRepository.unLockWishCard(wishCardId);
+      io.emit('unblock', {id: wishCardId});
+      clearTimeout(blockedWishcardsTimer[wishCardId])
+    }
+
+    res.status(200).send({
+      success: true,
+      error: null,
+    });
+  } catch (error) {
+    handleError(res, 400, error);
+  }
+});
+
+// check if user has donated
+router.get('/status/:id', async (req, res) => {
+
+  try {
+    const wishCardId = req.params.id;
+
+    if (!req.session.user) return handleError(res, 400, 'User not found');
+
+    const user = await UserRepository.getUserByObjectId(req.session.user._id);
+    if (!user) return handleError(res, 400, 'User not found');
+
+    clearTimeout(blockedWishcardsTimer[wishCardId])
+    const wishCard = await WishCardRepository.getWishCardByObjectId(wishCardId);
+
+    queue.add({wishCardId,
+      userId: req.session.user._id,
+      url:wishCard.wishItemURL,
+      price: wishCard.wishItemPrice});
+
+    res.status(200).send({
+      success: true,
+      error: null,
+    });
+
+  } catch (error) {
+    handleError(res, 400, error);
+  }
+
+});
+
+
+queue.process(async (job, done) => {
+
+  const {wishCardId, userId, url, price} = job.data;
+
+  try {
+
+    let isDonated = true;
+    const wishListArray = /registryID\.1=(.*?)&.*?registryItemID.1=(.*?)&/gm.exec(url);
+    if(wishListArray) {
+
+      const wishListId = wishListArray[1];
+      const itemId = wishListArray[2];
+
+      if (wishListId) {
+        const scrapeResponse = await scrapeList(`https://www.amazon.com/hz/wishlist/ls/${wishListId}` );
+
+        logger.debug(scrapeResponse)
+        if (!scrapeResponse) isDonated = false;
+
+        if (Object.keys(scrapeResponse).length > 0) {
+          isDonated = !JSON.stringify(scrapeResponse).includes(itemId);
         }
 
-        const lockedWishCard = await WishCardRepository.lockWishCard(wishCardId, user._id);
-        io.emit('block', { id: wishCardId, lockedUntil: lockedWishCard.isLockedUntil });
+        if(isDonated) {
+
+          const wishCard = await WishCardRepository.getWishCardByObjectId(wishCardId);
+
+          wishCard.isDonated = true;
+          wishCard.save();
```

Comment:
Wishcard isn't used anywhere else from here on, would move the updating to the repository and only make the function call to update it, makes it reusable